TaskManager.java

```diff
@@ -0,0 +1,71 @@
+@Component
+public class TaskManager {
+    @PersistenceContext
+    private EntityManager entityManager;
+    private static TaskManager instance;
+    private LinkedList<Task> queue = new LinkedList<>();
+    private HashSet<Task> running = new HashSet<>();
+    private HashSet<Task> blocked = new HashSet<>();
+    private Thread checker = new Thread(() -> {
+        while (true) {
+            checkRun();
+        }
+    });
+
+    public TaskManager() {
+        instance = this;
+        checker.run();
+    }
+
+    public static TaskManager getInstance() {
+        return instance;
+    }
+
+    @Transactional
+    public void runTask(Task task) {
+        if (Collections.disjoint(task.getDependsOn(), running)) {
+            running.add(task);
+            task.setStatus(TaskStatus.Working);
+            new Thread(() -> task.runTask(entityManager)).run();
+            entityManager.persist(new TaskHistoryRecord("task " + task.getName() + " " + task.getSchedule() + " " +
+                    " started " + System.currentTimeMillis()));
+        } else {
+            blocked.add(task);
+            task.setStatus(TaskStatus.Blocked);
+            entityManager.persist(new TaskHistoryRecord("task " + task.getName() + " " + task.getSchedule() + " " +
+                    " blocked " + System.currentTimeMillis()));
+        }
+    }
+
+    @Transactional
+    public void taskFinished(Task task) {
```

Comment:
From the Transactional context calling run method it is not good idea. 
It can take long time and will not commit